postgres-native-sqldelight-driver/src/commonMain/kotlin/app/softwork/sqldelight/postgresdriver/PostgresNativeDriver.kt

```diff
@@ -3,36 +3,101 @@ package app.softwork.sqldelight.postgresdriver
 import app.cash.sqldelight.*
 import app.cash.sqldelight.db.*
 import kotlinx.cinterop.*
-import kotlinx.datetime.*
-import kotlinx.uuid.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
 import libpq.*
-import kotlin.time.*
-
-public class PostgresNativeDriver(private var conn: CPointer<PGconn>) : SqlDriver {
+import platform.posix.*
+import kotlin.coroutines.*
+import kotlin.time.Duration.Companion.ZERO
+
+public class PostgresNativeDriver(
+    private var conn: CPointer<PGconn>,
+    private val coroutineScope: CoroutineScope?
+) : SqlDriver {
```

Comment:
It seems `CoroutineScope` cannot be `null`, or the `listener` support will be implicitly disabled.
What about turning this into `Closeable` and creating a custom `CoroutineScope`? That would be inline with `DataSource : Closeable` 🤔

Another question then would be though, which `CoroutineContext` would need to be used for the `CoroutineScope`? Can we safely default to `Dispatchers.IO`, or custom `CoroutineDispatcher`?

Reply:
`newFixedThreadPoolContext` is also available for `native` to create a "custom IO pool".

Reply:
Thanks, I do use `newSingleThreadContext` now. You always have to provide a `coroutineScope` if you want to get local or remote notifications. Reason: the flow extension also requires a coroutine context, so making this explicit is better (and now the local notifications use a flow too).